common/caching/cache.go

```diff
@@ -0,0 +1,8 @@
+package caching
+
+import "github.com/matrix-org/gomatrixserverlib"
+
+type Cache interface {
+	GetRoomVersion(roomId string) (gomatrixserverlib.RoomVersion, bool)
+	StoreRoomVersion(roomId string, roomVersion gomatrixserverlib.RoomVersion)
+}
```

Comment:
Feels like we're prematurely interfacing this just in case we have caches that are not just in-memory. I'd prefer it if we didnt until we actually wanted another cache mechanism.

Reply:
Honestly a bigger part of this is because I wanted one single place where we can look to see exactly what we have caching functions for.